sources/host-ctr/cmd/host-ctr/registry.go

```diff
@@ -58,10 +58,12 @@ func registryHosts(registryConfig *RegistryConfig, authorizerOverride *docker.Au
 			authConfig runtime.AuthConfig
 		)
 		// Set up endpoints for the registry
-		if _, ok := registryConfig.Mirrors[host]; ok {
-			endpoints = registryConfig.Mirrors[host].Endpoints
-		} else {
-			endpoints = registryConfig.Mirrors["*"].Endpoints
+		if registryConfig != nil {
+			if _, ok := registryConfig.Mirrors[host]; ok {
+				endpoints = registryConfig.Mirrors[host].Endpoints
+			} else {
+				endpoints = registryConfig.Mirrors["*"].Endpoints
+			}
```

Comment:
Haven't you created a new problem that `endpoints` can now be unset? I think we probably want the else clause to be for the first if statement:

```suggestion
		if registryConfig != nil {
			if _, ok := registryConfig.Mirrors[host]; ok {
				endpoints = registryConfig.Mirrors[host].Endpoints
			} 
		} else {
		     endpoints = registryConfig.Mirrors["*"].Endpoints

```

Reply:
Likewise, another good candidate for a comment. We start with an empty array of endpoints. Then we may replace that empty array with a non-empty one if registryConfig has an override. Then we append `docker.defaultHost(host)` to the array. The endpoints array always has at least that one `docker.defaultHost()` entry.